airlock/users.py

```diff
@@ -39,13 +46,24 @@ def to_dict(self):
     def has_permission(self, workspace_name):
         return (
             # Output checkers can view all workspaces
-            self.output_checker or self.can_create_request(workspace_name)
+            # Authors can view all workspaces they have access to (regardless
+            # of archive or project ongoing status)
+            self.output_checker or workspace_name in self.workspaces
         )
 
-    def can_create_request(self, workspace_name):
-        # Only users with explict access to the workspace can create release
+    def can_action_request(self, workspace_name):
+        # Only users with explict access to the workspace can create/modify release
         # requests.
-        return workspace_name in self.workspaces
+        if workspace_name not in self.workspaces:
```

Comment:
No, probably not - at the time I think I thought I'd reuse them

Reply:
So I think I actually did this because the workspace view just wants to know if the user can action the request, but doesn't need to raise the exception there, whereas in the couple of places in business_logic.py, it does raise the exception with different messages. However, I think it's better just raising the ActionDenied as you suggest, and `try/except`ing it in the workspace view.